csrc/sampling.cu

```diff
@@ -25,6 +25,33 @@
 
 using namespace flashinfer;
 
+void sampling_from_logits(at::Tensor logits, at::Tensor output,
+                          std::optional<at::Tensor> maybe_indices, bool deterministic,
+                          std::optional<at::Generator> gen_) {
+  CHECK_INPUT(logits);
+  auto device = logits.device();
+  CHECK_DIM(2, logits);  // logits: (batch_size, vocab_size)
+  unsigned int batch_size = output.size(0);
+  unsigned int vocab_size = logits.size(1);
+
+  uint64_t philox_seed, philox_offset;
+  auto gen = at::get_generator_or_default<at::CUDAGeneratorImpl>(
+      gen_, at::cuda::detail::getDefaultCUDAGenerator());
+  std::lock_guard<std::mutex> lock(gen->mutex_);
+  at::PhiloxCudaState rng_engine_inputs = gen->philox_cuda_state(batch_size);
```

Comment:
Set `increment` to `batch_size * vocab_size` because we have to generate large number of uniform numbers.
```suggestion
  at::PhiloxCudaState rng_engine_inputs = gen->philox_cuda_state(batch_size * vocab_size);
```

See https://github.com/pytorch/pytorch/blob/b34146a093ed868d0cb7de63c094f847cf82ad0b/aten/src/ATen/cuda/CUDAGeneratorImpl.cpp#L432-L442 for the explaination of `increment`.